### PR TITLE
refactor: Decouple the module from @appium/gulp-plugins

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Similar to https://github.com/appium/node-teen_process/pull/84

### CODE REVIEW

**Review by KazuCocoa — APPROVED** [review]
seems good to me, so the build can be done by babel instead of gulp. gotcha.

**Review by jlipps — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -16,9 +16,10 @@
   "bugs": {
     "url": "https://github.com/appium/node-simctl/issues"
   },
-  "engines": [
-    "node"
-  ],
+  "engines": {
+    "node": ">=14",
+    "npm": ">=6"
```
Comment (boneskull) [inline package.json:21]: should be 8

File: package.json [hunk 2]
```diff
@@ -44,33 +45,53 @@
...
-    "clean": "rm -rf node_modules && rm -f package-lock.json && npm install",
-    "prepare": "gulp prepublish",
-    "test": "gulp once",
-    "watch": "gulp watch",
-    "build": "gulp transpile",
-    "e2e-test": "gulp e2e-test",
-    "mocha": "mocha",
-    "coverage": "gulp coveralls",
+    "build": "rm -rf build/lib/* && babel --out-dir=build/lib lib",
+    "build-test": "rm -rf build/test/* && babel --out-dir=build/test test --copy-files",
```
Comment (boneskull) [inline package.json:49]: you don't need to build the tests since mocha runs w/ @babel/register. just run mocha against test/ instead of build/test

**Review by boneskull — CHANGES_REQUESTED** [review]
just a couple issues here otherwise looks good

File: package.json (hunk 1 above)
Comment (mykola-mokhnach) [reply package.json:21]: changed

File: package.json (hunk 2 above)
Comment (mykola-mokhnach) [reply package.json:49]: fixed

Comment (boneskull) [reply package.json]: `npm run build` is still happening before tests. does it need to?  I suppose if there's some sort of command-line e2e test going on

Comment (mykola-mokhnach) [reply package.json]: I was just not sure if mocha babel plugin also covers that. My assumption was it only automatically compiles tests, but not src. Is that assumption right?

Comment (boneskull) [reply package.json]: for posterity

image

**Pull request merged**